fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/NodePropertiesTools.java

```diff
@@ -203,9 +203,18 @@ public void removeNodeProperty(final Node node, final String propertyName, final
             if (property.isMultiple()) {
                 final AtomicBoolean remove = new AtomicBoolean();
                 final Value[] newValues = stream(node.getProperty(propertyName).getValues()).filter(v -> {
-                    if (v.equals(valueToRemove)) {
-                        remove.set(true);
-                        return false;
+                    String strVal = "";
+                    try {
+                        strVal = v.getString();
+
+                        if (strVal.equals(valueToRemove.getString())) {
+                            remove.set(true);
+                            return false;
+                        }
+
+                    } catch (RepositoryException e) {
```

Comment:
If the remove fails without effect in the API, the repository will not be in the situation in which the user tried to put it, but this will happen quietly. The user will continue blithely with his or her life, while the repository marches off in a different and strange direction. At some later time, the user will discover to his or her grimace- and heartburn-inducing chagrin that the giant literal he or she thought banished remains in place and is now causing confusion, dismay, existential angst, and unexpected fixity results.

I think that failure should stop computation and throw an exception back into the API.